pkg/core/blockchain.go

```diff
@@ -399,6 +407,54 @@ func (bc *Blockchain) init() error {
 	return bc.updateExtensibleWhitelist(bHeight)
 }
 
+// JumpToState is an atomic operation that changes Blockchain state to the one
+// specified by the state sync point p. All the data needed for the jump must be
+// collected by the state sync module.
+func (bc *Blockchain) JumpToState(p uint32) error {
+	bc.lock.Lock()
+	bc.log.Info("jumping to state sync point", zap.Uint32("state sync point", p))
+
+	block, err := bc.dao.GetBlock(bc.headerHashes[p])
+	if err != nil {
+		bc.lock.Unlock()
+		return fmt.Errorf("failed to get current block: %w", err)
+	}
+	err = bc.dao.StoreAsCurrentBlock(block, nil)
+	if err != nil {
+		bc.lock.Unlock()
+		return fmt.Errorf("failed to store current block: %w", err)
+	}
+	bc.blockHeight = p
+	bc.persistedHeight = p
+
+	block, err = bc.dao.GetBlock(bc.headerHashes[p+1])
+	if err != nil {
+		bc.lock.Unlock()
+		return fmt.Errorf("failed to get block to init MPT: %w", err)
+	}
+	if err = bc.stateRoot.JumpToState(&state.MPTRoot{
```

Comment:
By the end of the state sync process, there's no stateroot for the height P in the storage, there're only MPT nodes for this P state. It is done in order not to mess up the actual statemodule state (which is likely on the lower height). And `JumpToState` is aimed to *store* P-th stateroot.

> Maybe it can be replaced by new one initied at the height needed
Yes, it can, but then we either need to store the stateroot in a separate method or modify stateroot's `Init(...)`  method.